### PR TITLE
setup basics componets, pages and structure

### PR DESCRIPTION
**Pull request by EydenVillanueva**
## First Deliverable

List of changes:
- Installed required packages (styles-components, react-icons)
- Creation of all required components and pages.
- Basic styling using styles-components
- Consume mocks json data for displaying.

Link to de deployment visualization:
https://629d52945e71672afda07974--spiffy-gecko-0412b7.netlify.app/

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -9,7 +9,10 @@
     "eslint": "^7.32.0",
     "react": "^17.0.2",
     "react-dom": "^17.0.2",
+    "react-icons": "^4.4.0",
```
Comment (buty619) [inline package.json]: i don't think is a good idea install, this huge library just to use some few icons i prefer if you use it as svg react component or as image whit css.

File: src/components/categories/Categories.jsx [hunk 2]
```diff
@@ -0,0 +1,53 @@
+import React from "react";
+import PropTypes from "prop-types";
+import styled from "styled-components";
+
+import Category from "./Category";
+
+const Styles = styled.div`
```
Comment (buty619) [inline src/components/categories/Categories.jsx:7]: please check the identation of all files, please always handle 2 spaces identation.

File: src/components/categories/Categories.jsx [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+	h3 {
+		font-size: 2.4em;
+		font-weight: lighter;
+		margin-top: 2em;
+	}
+`
+
+
+const Categories = props => {
+	const {categories} = props;
```
Comment (buty619) [inline src/components/categories/Categories.jsx]: you can destructor this on the input props of the arrow function, ej `const Categories = ({categories}) => {`

File: src/components/categories/Categories.jsx [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+	
+	const renderCategories = () => {
```
Comment (buty619) [inline src/components/categories/Categories.jsx]: i think we is better to make the `categories.map(..)` on the JSX and avoid indirections.

File: src/components/categories/Category.jsx [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+  }
+
+  .container {
+    padding: 2px 16px;
+  }
+`
+
+
+const Category = props => {
+	const {properties} = props;
```
Comment (buty619) [inline src/components/categories/Category.jsx]: same comment as above

File: src/components/categories/Category.jsx [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+	
+	const renderCategoryCard = () => {
```
Comment (buty619) [inline src/components/categories/Category.jsx]: same comment as above, please avoid indirections.

File: src/components/header/SearchField.jsx [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+const Styles = styled.div`
+  input {
+    padding: 6px;
+    margin-top: 8px;
+    font-size: 17px;
+    border: none;
+  }
+`
+const SearchField = props => {
+	const { changeSearchField } = props;
```
Comment (buty619) [inline src/components/header/SearchField.jsx]: same comment as above

File: src/components/products/Product.jsx [hunk 8]
```diff
@@ -0,0 +1,108 @@
...
+    color: white;
+    background-color: #aaad61;
+	  border-radius: 10%;
+    padding: 5px;
+  }
+`
+
+
+const Product = props => {
+	const { properties } = props;
```
Comment (buty619) [inline src/components/products/Product.jsx]: same comment as above

File: src/components/products/Product.jsx [hunk 9]
```diff
@@ -0,0 +1,108 @@
...
+	
+	const renderProductCard = () => {
```
Comment (buty619) [inline src/components/products/Product.jsx]: same comment as above

File: src/components/slider/Slider.jsx [hunk 10]
```diff
@@ -0,0 +1,163 @@
...
+	
+`
+
+const Slider = () => {
+	
+	const { results: slidesData } = dataSlider;
+	const [slideIndex, setSlideIndex] = useState(1);
+	
+	
+	const nextSlide = () => {
```
Comment (buty619) [inline src/components/slider/Slider.jsx]: i think this could be his own component

File: src/components/slider/Slider.jsx [hunk 11]
```diff
@@ -0,0 +1,163 @@
...
+		if (slideIndex !== slidesData.length) {
+			setSlideIndex(slideIndex + 1);
+		} else if (slideIndex === slidesData.length) {
+			setSlideIndex(1);
+		}
+	}
+	
+	const prevSlide = () => {
```
Comment (buty619) [inline src/components/slider/Slider.jsx]: i think this could be his own component

**Review by buty619 — CHANGES_REQUESTED** [review]

File: package.json (hunk 1 above)
Comment (EydenVillanueva) [reply package.json]: Done

File: src/components/slider/Slider.jsx [hunk 12]
```diff
@@ -0,0 +1,157 @@
...
+  
+  const prevSlide = () => {
+    if (slideIndex === 1) {
+      setSlideIndex(slidesData.length);
+    } else if (slideIndex !== 1) {
+      setSlideIndex(slideIndex - 1);
+    }
+  }
+  
+  const renderSlides = () => {
```
Comment (buty619) [inline src/components/slider/Slider.jsx]: como te comentaba esta indireccion creo que no es conveniente, por fa corrígela ¡

File: src/pages/Content.jsx [hunk 13]
```diff
@@ -0,0 +1,13 @@
+import React from 'react';
+
+import Home from "./Home";
+
+const Content = props => {
```
Comment (buty619) [inline src/pages/Content.jsx:5]: te falto arreglar la identación de este file

File: src/pages/Footer.jsx [hunk 14]
```diff
@@ -0,0 +1,11 @@
+import React from 'react';
+
+const Footer = props => {
+	return (
+		<div>
```
Comment (buty619) [inline src/pages/Footer.jsx]: te falto arreglar la identación de este file

File: src/pages/Home.jsx [hunk 15]
```diff
@@ -0,0 +1,24 @@
...
+import rawCategories from "../mocks/en-us/product-categories.json";
+import rawProducts from "../mocks/en-us/products.json";
+
+
+const Home = () => {
+	const { results: categoryList } = rawCategories;
+	const { results: productList } = rawProducts;
+	
+	return (
+		<>
```
Comment (buty619) [inline src/pages/Home.jsx]: te falto arreglar la identación de este file

File: src/pages/Layout.jsx [hunk 16]
```diff
@@ -0,0 +1,17 @@
+import Content from "./Content";
+import Header from "../components/header/Header";
+import React from "react";
+import Footer from "./Footer";
+
+function Layout() {
+	
+	return (
+		<div className="App">
```
Comment (buty619) [inline src/pages/Layout.jsx]: te falto arreglar la identación de este file

File: src/utils/constants.js [hunk 17]
```diff
@@ -1 +1,22 @@
-export const API_BASE_URL = 'https://wizeline-academy.cdn.prismic.io/api/v2';
\ No newline at end of file
+export const API_BASE_URL = 'https://wizeline-academy.cdn.prismic.io/api/v2';
+
+const size = {
```
Comment (buty619) [inline src/utils/constants.js:3]: te falto arreglar la identación de este file

File: src/utils/truncate.js [hunk 18]
```diff
@@ -0,0 +1,5 @@
+const truncate = (toTruncate, maxLength) =>
+	(toTruncate.length > maxLength ? `${toTruncate.substring(0, maxLength)}...` : toTruncate);
```
Comment (buty619) [inline src/utils/truncate.js]: te falto arreglar la identación de este file

**Review by buty619 — CHANGES_REQUESTED** [review]

**Review by buty619 — APPROVED** [review]
🤟 excelente trabajo ¡¡ gracias por todas las modificaciones ¡¡

**Pull request merged**